server/src/main/java/org/apache/druid/server/coordination/ServerManager.java

```diff
@@ -117,7 +117,7 @@ public ServerManager(
 
     this.cacheConfig = cacheConfig;
     this.segmentManager = segmentManager;
-    this.joinableFactory = joinableFactory;
+    this.joinableFactoryWrapper = new JoinableFactoryWrapper(joinableFactory);
```

Comment:
`JoinableFactoryWrapper` still doesn't seem quite right. These walkers all take a `JoinableFactory` just to make a `JoinableFactoryWrapper` in the constructor. `JoinableFactoryWrapper` doesn't seem to have any state of its own, why not just make it in the joinable module and inject it directly? Or maybe it didn't really need to change from static methods?

Regardless, it doesn't need to change in this PR, we can refactor this in the future, since it feels like maybe there is also another refactor lurking that I haven't quite figured out in wrapping `CacheConfig`, `CachePopulator`, `Cache` (and maybe `CacheKeyManager` for brokers) into some tidy package to handle caching stuffs for walkers with implementations that do the right thing for where they are running.

Reply:
My primary reason for not using static methods, was easier mocking and unit-testing. I think injecting `JoinableFactoryWrapper` directly makes sense. Another option is to merge `MapJoinableFactory` and `JoinableFactoryWrapper` and cut down on a class. Both these classes are kind of providing a utility function on top of `JoinableFactory`.